### PR TITLE
DashboardSettings: Migrate Link Settings to React

### PR DESCRIPTION
**Pull request by jackw**
**What this PR does / why we need it**:
This PR migrates the link settings view from angular to react and introduces new form styles.

**Which issue(s) this PR fixes**:

Related Issue #21484

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by torkelo** [comment]
@jackw pushed an update to use delete button and IconButton for the link list

File: public/app/features/dashboard/components/LinksSettings/LinkSettingsEdit.tsx [hunk 1]
```diff
@@ -0,0 +1,132 @@
+import React, { useState } from 'react';
+import { css } from 'emotion';
+import { CollapsableSection, Button, TagsInput, Select, Field, Input, Switch } from '@grafana/ui';
+import { SelectableValue } from '@grafana/data';
+import { LinkSettingsMode } from '../DashboardSettings/LinksSettings';
+import { DashboardLink, DashboardModel } from '../../state/DashboardModel';
+
+const NEW_LINK = {
```
Comment (torkelo) [inline public/app/features/dashboard/components/LinksSettings/LinkSettingsEdit.tsx:8]: this capitalized variable naming we usually only use sparingly for value constants (numbers & strings)

**Comment by torkelo** [comment]
@jackw sorry, can't help myself, pushed some more UI improvements to the form (Guide is to use checkbox for options like this where you do not see a visible change right away)

**Review by torkelo — APPROVED** [review]
Looks great!

**Comment by torkelo** [comment]
See that this is still not merged, this seemed super close 12 days ago

File: public/app/features/dashboard/components/LinksSettings/LinkSettingsEdit.tsx (hunk 1 above)
Comment (jackw) [reply public/app/features/dashboard/components/LinksSettings/LinkSettingsEdit.tsx]: Sorry, old habits. Have camel cased them. 👍

**Comment by jackw** [comment]
I've taken this out of draft, added tests and tidied up based on your feedback @torkelo. I'm hoping to merge it tomorrow.

**Comment by torkelo** [comment]
@jackw a merge with master should fix the lint error

**Review by torkelo — APPROVED** [review]

**Pull request merged**